services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/fsm/FlowValidationFsm.java

```diff
@@ -0,0 +1,190 @@
+    private final FlowValidationRequest request;
+    private final FlowValidationHubCarrier carrier;
+    private final PersistenceManager persistenceManager;
+    private String flowId;
+    private FlowValidationService service;
+    private int awaitingRequests;
+    private List<SwitchFlowEntries> recievedRules = new ArrayList<>();
+    private List<FlowValidationResponse> response;
+
+    public FlowValidationFsm(FlowValidationHubCarrier carrier, String key, FlowValidationRequest request,
+                             PersistenceManager persistenceManager) {
+        this.carrier = carrier;
+        this.key = key;
+        this.request = request;
+        this.persistenceManager = persistenceManager;
+    }
+
+    /**
+     * FSM builder.
+     */
+    public static StateMachineBuilder<FlowValidationFsm, FlowValidationState, FlowValidationEvent, Object> builder() {
+        StateMachineBuilder<FlowValidationFsm, FlowValidationState, FlowValidationEvent, Object> builder =
+                StateMachineBuilderFactory.create(
+                        FlowValidationFsm.class,
+                        FlowValidationState.class,
+                        FlowValidationEvent.class,
+                        Object.class,
+                        FlowValidationHubCarrier.class,
+                        String.class,
+                        FlowValidationRequest.class,
+                        PersistenceManager.class);
+
+        builder.onEntry(INITIALIZED).callMethod("initialized");
+        builder.externalTransition().from(INITIALIZED).to(RECEIVE_RULES).on(NEXT)
+                .callMethod("receiveRules");
+        builder.internalTransition().within(RECEIVE_RULES).on(RULES_RECEIVED).callMethod("receivedRules");
+
+        builder.externalTransition().from(RECEIVE_RULES).to(FINISHED_WITH_ERROR).on(ERROR)
+                .callMethod(FINISHED_WITH_ERROR_METHOD_NAME);
+        builder.externalTransition().from(RECEIVE_RULES).to(VALIDATE_FLOW).on(NEXT)
+                .callMethod("validateFlow");
+
+        builder.externalTransition().from(VALIDATE_FLOW).to(FINISHED_WITH_ERROR).on(ERROR)
+                .callMethod(FINISHED_WITH_ERROR_METHOD_NAME);
+        builder.externalTransition().from(VALIDATE_FLOW).to(FINISHED).on(NEXT)
+                .callMethod(FINISHED_METHOD_NAME);
+
+        return builder;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    protected void initialized(FlowValidationState from, FlowValidationState to,
+                               FlowValidationEvent event, Object context) {
+        log.info("Key: {}; FSM initialized", key);
+        flowId = request.getFlowId();
+        service = new FlowValidationService(persistenceManager);
+    }
+
+    protected void receiveRules(FlowValidationState from, FlowValidationState to,
+                                FlowValidationEvent event, Object context) {
+        log.info("Key: {}; Send commands to get rules on the switches", key);
```

Comment:
Please do not use info log level for everything, you might use debug level for such purposes. Please correct me if I am wrong